cfme/tests/services/test_provision_stack.py

```diff
@@ -234,21 +244,54 @@ def test_remove_template_provisioning(setup_provider, provider, provisioning, ca
                   dialog=dialog_name_new, orch_template=template.template_name,
                   provider_type=provider.name)
     catalog_item.create()
-    if provider.type == 'ec2':
-        stack_data = {
-            'stack_name': "stack" + fauxfactory.gen_alphanumeric()
-        }
-    elif provider.type == 'openstack':
-        stack_data = {
-            'stack_name': "stack" + fauxfactory.gen_alphanumeric()
-        }
+    stackname = "test-" + fauxfactory.gen_alphanumeric()
+    stack_data = {'stack_name': stackname}
     service_catalogs = ServiceCatalogs("service_name", stack_data)
     service_catalogs.order_stack_item(catalog.name, catalog_item)
     flash.assert_no_errors()
-    template.delete()
+    template.delete_all_templates()
     row_description = 'Provisioning Service [{}] from [{}]'.format(item_name, item_name)
     cells = {'Description': row_description}
-    wait_for(lambda: requests.go_to_request(cells), num_sec=500, delay=20)
+    wait_for(lambda: requests.go_to_request_page(cells), num_sec=500, delay=20)
     row, __ = wait_for(requests.wait_for_request, [cells, True],
       fail_func=requests.reload, num_sec=1000, delay=20)
     assert row.last_message.text == 'Service_Template_Provisioning failed'
+
+
+@pytest.mark.uncollectif(lambda: version.current_version() < '5.5')
+def test_retire_stack(setup_provider, provider, provisioning,
+                      set_grid_view, dialog, catalog, request):
+    """Tests stack provisioning
+
+    Metadata:
+        test_flag: provision
+    """
+    # dialog_name, template_name = dialog
+    dialog_name_new = "dialog_" + fauxfactory.gen_alphanumeric()
+    template_type = provisioning['stack_provisioning']['template_type']
+    template = OrchestrationTemplate(template_type=template_type,
+                                     template_name=fauxfactory.gen_alphanumeric())
+    template.delete_all_templates()
+    template.create(METHOD_TORSO + "       ")
+    template.create_service_dialog_from_template(dialog_name_new, template.template_name)
+
+    item_name = fauxfactory.gen_alphanumeric()
+    catalog_item = CatalogItem(item_type="Orchestration", name=item_name,
+                  description="my catalog", display_in=True, catalog=catalog.name,
+                  dialog=dialog_name_new, orch_template=template.template_name,
+                  provider_type=provider.name)
+    catalog_item.create()
+    stackname = "test-" + fauxfactory.gen_alphanumeric()
+    stack_data = {'stack_name': stackname}
+
+    service_catalogs = ServiceCatalogs("service_name", stack_data)
+    service_catalogs.order_stack_item(catalog.name, catalog_item)
+    flash.assert_no_errors()
```

Comment:
Shouldn't this flash assert be in the ```service_catalog.order_stack_item``` function? generally that's where we assert now